src/collections/componentSet.ts

```diff
@@ -721,7 +739,7 @@ export class ComponentSet extends LazyCollection<MetadataComponent> {
 
 const sourceKey = (component: SourceComponent): string => {
   const { fullName, type, xml, content } = component;
-  return `${type.name}${fullName}${xml ?? ''}${content ?? ''}`;
+  return `${type.name}${fullName}${xml ?? ''}${content ?? ''}${component.isMarkedForDelete()}`;
```

Comment:
If this really is needed I think it should be more descriptive than appending `false` or `true`.  I'm hopeful that we can rely on the source components being accurate and assigning them into the correct component sets on add.